### PR TITLE
Data types

### PR DESCRIPTION
**Pull request by ad-prince**
- [x] Do the tabs work? I wonder if we should use dropdowns to make it look less busy.
- [x] Check the content - we could go into more detail in some places, is that needed?
- [x] Is it in the right place? Currently in Spark Overview.

### CODE REVIEW

**Review by robertswh — COMMENTED** [review]
A few comments to check. With regards to where to put it in the book, I think it's works where it is, we can review later. And regarding the tabs, maybe we should ask a few people what they think works best? There's arguments both ways so probably worth getting other people's opinions.

File: ons-spark/spark-overview/data-types.md [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+Rows: ??
+Columns: 4
+Database: spark_connection
+$ incident_number   <chr> "80771131", "141817141", "143166-22102016", "4305114…
+$ date_time_of_call <chr> "25/06/2013 07:47", "22/10/2014 17:39", "22/10/2016 …
+$ cal_year          <int> 2013, 2014, 2016, 2014, 2013, 2012, 2010, 2018, 2015…
+$ fin_year          <chr> "2013/14", "2014/15", "2016/17", "2014/15", "2012/13…
+```
+````
+CSV files do not have any schema attached to them. There are two options for determining the data types in a DataFrame when the source data is a CSV file: use `inferSchema`/`infer_schema`, or supply a schema directly with the `schema`/`columns` option when reading the data in.
```
[the remark this answers is not in the document]
Comment (ad-prince) [reply ons-spark/spark-overview/data-types.md:283]: Have added a small clarification to this.

File: ons-spark/spark-overview/data-types.md [hunk 2]
```diff
@@ -0,0 +1,479 @@
...
+Rows: ??
+Columns: 4
+Database: spark_connection
+$ incident_number   <chr> "139091", "275091", "2075091", "2872091", "3553091",…
+$ date_time_of_call <chr> "01/01/2009 03:01", "01/01/2009 08:51", "04/01/2009 …
+$ cal_year          <int> 2009, 2009, 2009, 2009, 2009, 2009, 2009, 2009, 2009…
+$ fin_year          <chr> "2008/09", "2008/09", "2008/09", "2008/09", "2008/09…
+```
+````
+In PySpark, using Data Definition Language (DDL) to define a schema is generally quicker and easier. You may be familiar with DDL when creating database tables with SQL. Just us the names of the columns followed by their data type and then separated with commas. For ease of reading it is better to use a multi-line string and put each entry on a new line. Remember that multi-line strings in Python need to be opened and closed with `"""`.
```
[the remark this answers is not in the document]
Comment (ad-prince) [reply ons-spark/spark-overview/data-types.md:386]: Done

**Comment by ad-prince** [comment]
Re: tabs - I say merge it in now and can always change it later. For now I'm focussed on the content.

**Pull request merged**